src/hooks/useCalculation.ts

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from 'react';
+import type { UnitBoxes } from '../types';
+import { useStateValue } from '../state/state';
+import { ARDataPriceEstimator } from '../utils';
+
+/** ARDataPriceEstimator instance, will fire off initial fetch calls when constructed */
+const arDataPriceEstimator = new ARDataPriceEstimator();
+
+const winstonToArValue = 0.000_000_000_001;
+
+interface previousValues {
+	bytes: number;
+	fiat: number;
+	ar: number;
+}
+
+/**
+ * Use calculation hook listens to changes to the global unitBox state. If a unitBox
+ * changes, it will re-calculate the boxes calculate the other boxes. Then, update
+ * the unitBox state with one dispatch call
+ *
+ * The hook is intended to be used once in the Calculator component
+ */
+export default function useCalculation(): void {
+	const [{ unitBoxes }, dispatch] = useStateValue();
+	const [sendingCalculation, setSendingCalculation] = useState(false);
+
+	// Save previous  values for determining if a value has changed
+	const [prevUnitVals, setPrevUnitVals] = useState<previousValues>({
+		bytes: unitBoxes.ar.value,
+		fiat: unitBoxes.fiat.value,
+		ar: unitBoxes.ar.value
+	});
+
+	/** @TODO Conversion will come from fiatToArData[unitBoxes.fiat.currUnit] */
+	const currentFiatToWinstonConversion = 0.000_000_000_15;
```

Comment:
A few things here:
1) I worry about the precision we'll be able to get when converting from other currencies to Winston. We'll likely want to convert from other currencies to AR, THEN convert to Winston to try and preserve as much precision as possible. I haven't fully thought through that though.
2) Please rename to `fiatPerAR`. That way we can think about the multiplications in code from a "dimensional analysis" standpoint, i.e. (AR)*(fiat/AR) = fiat.